tests/Unit/OutputFormatterTest.php

```diff
@@ -297,4 +299,257 @@ public function implodeWithIncreaseLevelTrueIncreasesIndentationLevelForRenderin
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithEmptyCommentableAndRenderCommentsEnabledReturnsEmptyString(): void
+    {
+        $this->outputFormat->setRenderComments(true);
+        $this->outputFormat->setSpaceBetweenBlocks(' between-space ');
+        $this->outputFormat->setSpaceAfterBlocks(' after-space ');
+
+        $commentable = $this->createMock(Commentable::class);
+        $commentable->method('getComments')->willReturn([]);
+
+        $result = $this->subject->comments($commentable);
+
+        self::assertSame('', $result);
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithCommentableWithOneCommentAndRenderCommentsDisabledReturnsEmptyString(): void
+    {
+        $this->outputFormat->setRenderComments(false);
+        $this->outputFormat->setSpaceBetweenBlocks(' between-space ');
+        $this->outputFormat->setSpaceAfterBlocks(' after-space ');
+
+        $commentText = 'This is a comment.';
+        $comment = new Comment($commentText);
+        $commentable = $this->createMock(Commentable::class);
+        $commentable->method('getComments')->willReturn([$comment]);
+
+        $result = $this->subject->comments($commentable);
+
+        self::assertSame('', $result);
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithCommentableWithOneCommentRendersComment(): void
+    {
+        $this->outputFormat->setRenderComments(true);
+        $this->outputFormat->setSpaceBetweenBlocks(' between-space ');
+        $this->outputFormat->setSpaceAfterBlocks(' after-space ');
+
+        $commentText = 'This is a comment.';
+        $comment = new Comment($commentText);
+        $commentable = $this->createMock(Commentable::class);
+        $commentable->method('getComments')->willReturn([$comment]);
+
+        $result = $this->subject->comments($commentable);
+
+        self::assertStringContainsString('/*' . $commentText . '*/', $result);
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithCommentableWithOneCommentPutsSpaceAfterBlocksAfterRenderedComment(): void
+    {
+        $this->outputFormat->setRenderComments(true);
+        $this->outputFormat->setSpaceBetweenBlocks(' between-space ');
+        $afterSpace = ' after-space ';
+        $this->outputFormat->setSpaceAfterBlocks($afterSpace);
+
+        $commentText = 'This is a comment.';
+        $comment = new Comment($commentText);
+        $commentable = $this->createMock(Commentable::class);
+        $commentable->method('getComments')->willReturn([$comment]);
+
+        $result = $this->subject->comments($commentable);
+
+        self::assertSame('/*' . $commentText . '*/' . $afterSpace, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithCommentableWithTwoCommentsPutsSpaceAfterBlocksAfterLastRenderedComment(): void
+    {
+        $this->outputFormat->setRenderComments(true);
+        $this->outputFormat->setSpaceBetweenBlocks(' between-space ');
```

Comment:
Not needed?

Reply:
I've reduced the tests now to be more focused.